src/battle_util.c

```diff
@@ -11419,6 +11430,25 @@ u16 GetUsedHeldItem(u32 battler)
     return gBattleStruct->usedHeldItems[gBattlerPartyIndexes[battler]][GetBattlerSide(battler)];
 }
 
+bool32 CantPickupItem(u32 battler)
+{
+    // Used by RandomUniformExcept() for RNG_PICKUP
+    if (battler == gBattlerAttacker && gBattleTypeFlags & (BATTLE_TYPE_TRAINER | BATTLE_TYPE_LINK))
```

Comment:
I'm worried about this part. How do you know the current value of `gBattlerAttacker`?

Reply:
Technically you would check against the `battler` in `ABILITY_PICKUP`, no?

Reply:
Right before the `switch (gLastUsedAbility)` just above `case ABILITY_PICKUP:` line 4969 (battle_util.c) `gBattlerAttacker = battler;`. From then until printing the message, gBattlerAttacker is never altered in any way.

Reply:
I see!